presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -584,6 +585,25 @@ public PlanWithProperties visitExplainAnalyze(ExplainAnalyzeNode node, Preferred
             return rebaseAndDeriveProperties(node, child);
         }
 
+        @Override
+        public PlanWithProperties visitAnalyzeFinish(AnalyzeFinishNode node, PreferredProperties context)
+        {
+            PlanWithProperties child = planChild(node, PreferredProperties.any());
+
+            // if the child is already a gathering exchange, don't add another
+            if ((child.getNode() instanceof ExchangeNode) && ((ExchangeNode) child.getNode()).getType().equals(GATHER)) {
```

Comment:
intellij shows me a warning here, "'equals()' called on Enum value"

Reply:
I didn't see the warning on mine. I think `equals` is fine because the implementation just calls `==` and we do the same thing for other operators (e.g. `visitTableFinish`)

Reply:
Admittedly, this isn't consistent across the codebase.

Reply:
Equals should not be called on an enum value.  You should use `==` instead.  The other places that do that are wrong and should be fixed in the future.